cmd/packetfence-forwarder-config-generator/main.go

```diff
@@ -0,0 +1,181 @@
+		Filter:             "udp and port 67 and ((udp[250:1] = 0x3) or (udp[250:1] = 0x5))",
+		DisableDNS:         true,
+		DNSDestinationHost: "",
+		DNSDestinationPort: 753,
+		DNSFilter:          "udp and port 53",
+	}
+}
+
+func main() {
+	fmt.Printf("!!! You can run this program anytime from %s !!!\n\n", os.Args[0])
+	//Set values to defaults. 0x3: DHCPREQUEST. 0x5: DHCPACK. Those are the only ones required by PacketFence to track and fingerprint devices from DHCP.
+	config := defaultConfig()
+	SelectInterface(&config)
+	SetupDHCPForwarding(&config)
+	SetupDNSForwarding(&config)
+	SaveConfig(&config, "Packetfence-Forwarder.toml")
+}
+
+func SelectInterface(c *Configuration) {
+
+	networkInterfaces := getInterfaces()
+
+	var interfaceChoose string
+	fmt.Printf("Index\t:\t Interface name\t\n")
+	for row, each := range networkInterfaces {
+		fmt.Printf("%d\t:\t %s\n", row, each.Description)
+	}
+
+	for {
+		fmt.Printf("\nPlease choose the number interface or all:  ")
+		if _, err := fmt.Scan(&interfaceChoose); err != nil {
+			fmt.Printf("Error. %v\n", err)
+			continue
+		}
+
+		if interfaceChoose == "all" {
+			c.ListeningDevice = ""
+			break
+		}
+
+		interfaceIndex, err := strconv.ParseInt(interfaceChoose, 10, 64)
+		if err == nil {
+			if 0 <= interfaceIndex && interfaceIndex < int64(len(networkInterfaces)) {
+				c.ListeningDevice = networkInterfaces[interfaceIndex].Name
+				break
+			}
+		}
+
+		fmt.Printf("!!! Choice out of possible range. Choose between 0 and %v or all !!!\n", len(networkInterfaces)-1)
+	}
+}
+
+func setupEnable(msg string) bool {
+	enable := ""
+	for {
+		fmt.Printf("\n%s y/n: ", msg)
+		if _, err := fmt.Scan(&enable); err != nil {
+			fmt.Printf("Error. %v\n", err)
+			continue
+		}
+
+		if enable[0] == 'Y' || enable[0] == 'y' {
+			return true
+		}
+
+		if enable[0] == 'N' || enable[0] == 'n' {
+			return false
+		}
+
+	}
+}
+
+func SetupDHCPForwarding(c *Configuration) {
+	if !setupEnable("Enable DHCP forwarding") {
+		c.DisableDHCP = true
+		return
+	}
+
+	c.DisableDHCP = false
+	setupHostAndPort("DHCP", &c.DestinationHost, &c.DestinationPort)
+}
+
+func SetupDNSForwarding(c *Configuration) {
+	if !setupEnable("Enable DNS forwarding") {
+		c.DisableDNS = true
+		return
+	}
+
+	c.DisableDNS = false
+	setupHostAndPort("DNS", &c.DNSDestinationHost, &c.DNSDestinationPort)
+}
+
+func setupHostAndPort(msg string, host *string, port *uint16) {
+	tmpStr := ""
+	tmpPort := -1
+	fmt.Printf("\nSetting up %s forward host and port\n", msg)
+	for {
+		fmt.Printf("To which IP will the selected %s traffic be forwarded to: ", msg)
+		if _, err := fmt.Scan(&tmpStr); err != nil {
+			fmt.Printf("Error. %v\n\n", err)
+			continue
+		}
+
+		testInput := net.ParseIP(tmpStr)
+		if testInput.To4() == nil {
+			fmt.Printf("!!! %v is not a valid hostv4 address !!!\n", tmpStr)
+			continue
+		} else {
+			*host = tmpStr
+			break
+		}
+	}
+
+	for {
+		fmt.Printf("To which UDP port will the selected %s traffic be forwarded to: ", msg)
```

Comment:
Why not adding a default port since you are using `config := defaultConfig()`
Like a default classic option